src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -440,11 +440,15 @@ export class SearchSource {
           await this.setDataFrame(dataFrameResponse.body as IDataFrame);
           return onResponse(searchRequest, convertResult(response as IDataFrameResponse));
         }
+        if ((response as IDataFrameResponse).type === DATA_FRAME_TYPES.POLLING) {
```

Comment:
nit: Can we combine both the if condition 
`(response as IDataFrameResponse).type === DATA_FRAME_TYPES.DEFAULT) || (response as IDataFrameResponse).type === DATA_FRAME_TYPES.POLLING)`
since the body of the both the condition are same.

Reply:
yes good catch. i put this here but i'm planning on removing this `type` difference